### PR TITLE
Option to disable badge animation

### LINKED ISSUE #89: [Feature request] Blinking icon is too distracting. Having an option to disable it might be nice.
Initial state:
initial state, only red squared icon
Alternate state:
alternate state, req square is covered by some more data, like the metric failing and the offset value

Currently, the badge is [designed](https://github.com/GoogleChrome/web-vitals-extension/blob/4d51b39e54a5199d9aa09fd6ae3c75fd7d03008b/src/bg/background.js#L240) to be blinking constantly.

It's a little bit too distracting when the Extension icon is "pinned" on Chrome's toolbar.

I think having the blinking as an option you can disable would be more friendly. Or just show the red icon with the badge (one of the current states).

### PR DESCRIPTION
**Pull request by tunetheweb**
Fixes #89 

This adds an option to disable the cycling of badge updates between overall status and each failing metric.

Note: The badge will still update on status change.

Open to better option name suggestions:

image

### CODE REVIEW

File: service_worker.js [hunk 1]
```diff
@@ -249,8 +252,18 @@ async function animateBadges(request, tabId) {
   const delay = 2000;
   // First badge overall perf
   badgeOverallPerf(request.passesAllThresholds, tabId);
+
   // If perf is poor, animate the sequence
   if (request.passesAllThresholds === 'POOR') {
+
+    // However, if user has turned this off, then leave it off.
+    // Note, we don't call animateBadges again so need a new metric event or a
```
Comment (brendankenny) [inline service_worker.js:260]:
```suggestion
    // TODO(tunetheweb): we don't call animateBadges again so need a new metric event or a
```
?

File: service_worker.js [hunk 2]
```diff
@@ -19,6 +19,8 @@ const CLS_THRESHOLD = 0.1;
 const FCP_THRESHOLD = 1800;
 const TTFB_THRESHOLD = 800;
 
+let optionsNoBadgeAnimation = false;
```
Comment (brendankenny) [inline service_worker.js:22]: doesn't the initial value of this need to be read from `chrome.storage.sync`?

File: service_worker.js [hunk 3]
```diff
@@ -287,3 +300,11 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
     sendResponse({tabId: sender.tab.id});
   }
 });
+
+// Listen for changes to noBadgeAnimation option
+function logStorageChange(changes, area) {
+  if (area === 'sync' && 'noBadgeAnimation' in changes) {
+    optionsNoBadgeAnimation = changes['noBadgeAnimation'].newValue;
```
Comment (brendankenny) [inline service_worker.js:307]: nit

```suggestion
    optionsNoBadgeAnimation = changes.noBadgeAnimation.newValue;
```

File: service_worker.js (hunk 2 above)
Comment (tunetheweb) [reply service_worker.js]: Good spot! Fixed in https://github.com/GoogleChrome/web-vitals-extension/pull/141/commits/010d88bc15b6f8ff6e048abb51bbf00d74f94fcd

File: service_worker.js (hunk 1 above)
Comment (tunetheweb) [reply service_worker.js:268]: I don't think this is a TODO so much as an observation.

If you're RED and not animating, and then re-enable animation it won't start animating immediately until a status changes or the page reload. I think that's fair enough, and not worth the complexity to reenable animations immediately. In the same way as the other options HUD, console logging don't activate immediately

WDYT?

File: service_worker.js (hunk 3 above)
Comment (tunetheweb) [reply service_worker.js]: Fixed in https://github.com/GoogleChrome/web-vitals-extension/commit/010d88bc15b6f8ff6e048abb51bbf00d74f94fcd

File: service_worker.js [hunk 4]
```diff
@@ -19,6 +19,16 @@ const CLS_THRESHOLD = 0.1;
 const FCP_THRESHOLD = 1800;
 const TTFB_THRESHOLD = 800;
 
+// Get the optionsNoBadgeAnimation value
+// Actual default is false but lets set to true
+// initially in get sync storage is slow
```
Comment (brendankenny) [inline service_worker.js:24]: "in case sync storage is slow"?

File: service_worker.js (hunk 1 above)
Comment (brendankenny) [reply service_worker.js:268]: That's fine, maybe reword the comment slightly then? I just suggested it because it currently sounds like a TODO :)

Adapting your comment here, maybe something like:
```js
    // Note: if optionsNoBadgeAnimation is flipped, it won't start (or stop)
    // animating immediately until a status change or page reload to avoid
    // having to check continually.
```
or whatever sounds good.

**Review by brendankenny — APPROVED** [review]
